src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -228,12 +229,32 @@ public static unsafe string GetRandomFileName()
             byte* pKey = stackalloc byte[KeyLength];
             Interop.GetRandomBytes(pKey, KeyLength);
 
-            const int RandomFileNameLength = 12;
             char* pRandomFileName = stackalloc char[RandomFileNameLength];
             Populate83FileNameFromRandomBytes(pKey, KeyLength, pRandomFileName, RandomFileNameLength);
             return new string(pRandomFileName, 0, RandomFileNameLength);
         }
 
+        /// <summary>
+        /// Returns a cryptographically strong random 8.3 string that can be
+        /// used as either a folder name or a file name.
+        /// </summary>
+        public static unsafe bool TryGetRandomFileName(Span<char> destination, out int charsWritten)
+        {
+            charsWritten = 0;
+
+            if (destination.Length < RandomFileNameLength)
+                return false;
+
+            byte* pKey = stackalloc byte[KeyLength];
+            Interop.GetRandomBytes(pKey, KeyLength);
+
+            char* pRandomFileName = stackalloc char[RandomFileNameLength];
+            Populate83FileNameFromRandomBytes(pKey, KeyLength, pRandomFileName, RandomFileNameLength);
```

Comment:
I'd change Populate83FileName to take Spans. It isn't perf critical, just don't want to add heap allocations.